packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -167,9 +167,8 @@ const ImageDialog = ({ handleClose }) => {
   const MediaLibraryDialog = components['media-library'];
 
   const insertImages = (images) => {
-    images.forEach((img) => {
-      const image = { type: 'image', image: img, children: [{ type: 'text', text: '' }] };
-      Transforms.insertNodes(editor, image);
+    images.forEach((image) => {
+      Transforms.setNodes(editor, { image });
```

Comment:
How does this work, the image node was already created like when we clicked it in the select dropdown and then we are updating the properties here?

Reply:
If so do you min just clarifying that with a comment? I'm just thinking someone might come along and think oh this should be insertNodes 🤦‍♂️

Reply:
yes for the single image it works that way, but I realised it wont work for multiple images, so lets not include in alpha I will fix it further.

Reply:
I feel like we should be creating the entire block here, and selecting Image shouldn't creating anything in Slate. Also because I think otherwise if we press "cancel" on the media library modal we end up with an empty image node

Reply:
right! toggle is the culprit